java/build/bazel/tests/integration/WorkspaceDriver.java

```diff
@@ -223,21 +225,18 @@ public void scratchExecutableFile(String path, Iterable<String> content) throws
   }
 
   private File writeToFile(String path, Iterable<String> content) throws IOException {
-    File dest = new File(workspace, path);
-    if (!dest.getParentFile().exists()) {
-      dest.getParentFile().mkdirs();
-    }
-    Files.write(dest.toPath(), String.join("\n", content).getBytes(StandardCharsets.UTF_8));
-    return dest;
+    Path dest = createParentDirectoryIfNotExists(path);
+    Files.write(dest, String.join("\n", content).getBytes(StandardCharsets.UTF_8));
+    return dest.toFile();
```

Comment:
Why do we need to do “toFile” here?

Reply:
It’s because scratchExecutableFile needs to set it as executable.
I think you can use Files to do it on a path